src/platform_impl/linux/util.rs

```diff
@@ -70,3 +74,47 @@ pub fn set_size_constraints<W: GtkWindowExt + WidgetExt>(
     geom_mask,
   )
 }
+
+pub struct WindowMaximizeProcess<W: GtkWindowExt + WidgetExt> {
+  window: W,
+  maximized: bool,
+  resizable: bool,
+  step: u8,
+}
+
+impl<W: GtkWindowExt + WidgetExt> WindowMaximizeProcess<W> {
+  pub fn new(window: W, maximized: bool, resizable: bool) -> Rc<RefCell<Self>> {
+    Rc::new(RefCell::new(Self {
+      window,
+      maximized,
+      resizable,
+      step: 0,
+    }))
+  }
+
+  pub fn next_step(&mut self) -> glib::ControlFlow {
+    match self.step {
+      0 => {
+        if self.maximized {
+          self.window.set_resizable(true);
+          self.step += 1;
+          glib::ControlFlow::Continue
+        } else {
+          //It should still revert to the previous resizable state.
+          self.step = 2;
+          glib::ControlFlow::Continue
+        }
```

Comment:
taking another look at the usage of `WindowMaximizeProcess` we seem to be only using it when `maximized` is `true` so we can simplify this to 
```suggestion
       self.window.set_resizable(true);
       self.step += 1;
       glib::ControlFlow::Continue
```

and we can remove `maximized` field from `WindowMaximizeProcess` since it will always be true

Reply:
Indeed, since all the code using `WindowMaximizeProcess` is wrapped in a `maximized` condition, the suggestion seems better.

Reply:
It works well. I will go ahead and revise the code with the suggested changes.

![maximize-update](https://github.com/user-attachments/assets/5f25c8e4-f939-4f22-a098-56af02d2a32d)